Scripts/Runtime/Entities/TaskSystem/TaskData/DataStream/Profiling/DataStreamProfilingSystem.cs

```diff
@@ -0,0 +1,60 @@
+using Anvil.Unity.DOTS.Jobs;
+using System.Collections.Generic;
+using Unity.Entities;
+
+namespace Anvil.Unity.DOTS.Entities.Tasks
+{
+#if DEBUG
+    /// <summary>
+    /// System that will handle populating profiling counters once per frame when guaranteed that all writing jobs
+    /// have completed.
+    /// </summary>
+    //TODO: #86 - Revisit with Entities 1.0 for "Create Before/After"
+    //TODO: #108 - Revisit to see if there is a better way to handle acquire/release. Or can we do it all in a job? 
+    [UpdateInGroup(typeof(PresentationSystemGroup), OrderLast = true)]
+    public partial class DataStreamProfilingSystem : AbstractAnvilSystemBase
+    {
+        private readonly List<AbstractDataStream> m_AllDataStreams;
+        
+        private TaskFlowGraph m_TaskFlowGraph;
+        private bool m_HasInitialized;
+
+        public DataStreamProfilingSystem()
+        {
+            m_AllDataStreams = new List<AbstractDataStream>();
+        }
+
+        protected override void OnCreate()
+        {
+            m_TaskFlowGraph = World.GetExistingSystem<TaskFlowSystem>().TaskFlowGraph;
+            base.OnCreate();
+        }
+
+        protected override void OnStartRunning()
+        {
+            if (m_HasInitialized)
+            {
+                return;
+            }
+
+            m_HasInitialized = true;
+
+            m_TaskFlowGraph.AddAllDataStreamsTo(m_AllDataStreams);
+            
+            base.OnStartRunning();
+        }
+
+        protected override void OnUpdate()
+        {
+            foreach (AbstractDataStream dataStream in m_AllDataStreams)
+            {
+                dataStream.AccessController.Acquire(AccessType.SharedRead);
+                
+                dataStream.PopulateProfiler();
+                
+                dataStream.AccessController.Release();
+            }
```

Comment:
Seems like you could build a native array of structs that contain `ProfilerData` structs. Go through each datastream and AcquireAsync. Then create a job that is given:
 - data stream
 - native array of profiler structs and index to write to (or a pointer to the right element) 

The job would pull the profiler data off of the datastream and write it to the correct element in the native array. Then the native array gets read when the profiler updates.